provisionersdk/scripts/bootstrap_linux.sh

```diff
@@ -43,6 +43,43 @@ if ! chmod +x $BINARY_NAME; then
 	exit 1
 fi
 
+haslibcap2() {
+	command -v setcap /dev/null 2>&1
+	command -v capsh /dev/null 2>&1
+}
+
+# Attempt to add CAP_NET_ADMIN to the agent binary. This allows us to increase
+# network buffers which improves network transfer speeds.
+if [ -n "${USE_CAP_NET_ADMIN}" ]; then
+	# If running as root, we don't need to do anything.
+	if [ "$(id -u)" -eq 0 ]; then
+		echo "Running as root, skipping setcap"
+		# Warn the user if root doesn't have CAP_NET_ADMIN.
+		if ! capsh --has-p=CAP_NET_ADMIN; then
+			echo "The root user doesn't have CAP_NET_ADMIN permission. " + \
+				"If running in Docker, add the capability to the container for " + \
+				"improved network performance."
+		fi
+
+	# If not running as root, make sure we have sudo perms and the 'setcap' binary
+	# exists.
+	elif sudo -nl && haslibcap2; then
+		# Make sure the root user has CAP_NET_ADMIN.
+		if sudo -n capsh --has-p=CAP_NET_ADMIN; then
+			sudo -n setcap CAP_NET_ADMIN=+ep ./$BINARY_NAME || true
+		else
+			echo "The root user doesn't have CAP_NET_ADMIN permission. " + \
+				"If running in Docker, add the capability to the container for " + \
+				"improved network performance."
+		fi
+
+	# If we're not running as root, can't sudo, and 'setcap' doesn't exist, we can't
+	# do anything.
+	else
+		echo "Unable to setcap agent binary. Missing passwordless sudo permissions or the 'setcap' binary."
```

Comment:
This makes it sound like only one is required, when you need both. Maybe say `To enable improved network performance, give the agent passwordless sudo permissions and the 'setcap' binary.`

Reply:
This should also say what it's for and what the security implications are